src/rust/bitbox02-rust/src/hash.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloc::vec::Vec;
+
+/// Implements the digest traits for Sha512 backing it with the wally_sha512 C function. This is
+/// done to avoid using a second sha512 implementation like `sha2::Sha512`, which bloats the binary
+/// by an additional ~12.7kB (at the time of writing).
+///
+/// This implementation accumulates the data to be hashed in heap, it does **not** hash in a
+/// streaming fashion, even when using `update()`.
+#[derive(Default, Clone)]
+pub struct Sha512(Vec<u8>);
```

Comment:
Maybe name it `Sha512Message`, `Sha512Data` or something? It took me a while to understand that it contains the message and not the digest while reading the code.

Reply:
I changed my mind, I suppose `Sha512` is a good name. It is just strange that it contains the message. But it is described in the comment.

I guess
```
pub struct Sha512 {
    message: Vec<u8>
}
```

would make it slightly clearer.

Reply:
Makes sense, I named the struct field as you suggested now.